main/src/library/pkger/SemVer.flix

```diff
@@ -0,0 +1,118 @@
+namespace SemVer {
+    // TODO: Consider adding pre-release identifier and build metadata
+    pub type alias SemVer = (Int, Int, Int)
+
+    pub enum Ord {
+        case Gt
+        case Eq
+        case Lt
+    }
+
+    pub def isWithinBounds(bound: SemVer, specific: SemVer): Bool = {
+        let majorIsZero = match bound {
+            case (major, _, _) => major == 0
+        };
+
+        if (!majorIsZero) {
+            match (specific, bound) {
+                // Only major versions allow backwards incompatibility
+                case ((major, _, _), (boundMajor, _, _)) => boundMajor == major
+            }
+        } else {
+            // TODO: Look only for specific version, or allow patch releases? Only specific for now!
+            match (specific, bound) {
+                case ((major, minor, _), (boundMajor, boundMinor, _)) => major == boundMajor && minor == boundMinor
+            }
+        }
+    }
+
+    pub def cmpSemVer(left: SemVer, right: SemVer): Ord = match (left, right) {
+        case ((lMajor, lMinor, lPatch), (rMajor, rMinor, rPatch)) => {
+            if (lMajor == rMajor) {
+                if (lMinor == rMinor) {
+                    if (lPatch == rPatch) {
+                        Eq
+                    } else if (lPatch > rPatch) {
+                        Gt
+                    } else {
+                        Lt
+                    }
+                } else if (lMajor > rMajor) {
+                    Gt
+                } else {
+                    Lt
+                }
+            } else if (lMajor > rMajor) {
+                Gt
+            } else {
+                Lt
+            }
+        }
+    }
+
+    pub def ppOrd(o: Ord): String = match o {
+        case Gt => "Gt",
+        case Eq => "Eq",
+        case Lt => "Lt"
+    }
+
+    namespace Parser {
+        def parseNumber(): Parser[Int] = Parser.mapOption(Parser.failOn(Parser.span(Char.isDigit), String.isEmpty), s -> Result.toOption(Int32.fromString(s)))
+
+        pub def parseSemVer(): Parser[SemVer.SemVer] = Parser.bind(parseNumber(), major -> {
+            Parser.bind(Parser.char('.') *> parseNumber(), minor -> {
+                Parser.bind(Parser.char('.') *> parseNumber(), patch -> {
+                    Parser.return((major, minor, patch))
+                })
+            })
+        })
+    }
+
+    namespace Test {
```

Comment:
Ok, I see you have some tests. I would put them in a separate file, so you have `SemVer.flix` and `TestSemVer.flix`.

Then if you add `@test`, you can run the compiler with `--test`.